db/index.go

```diff
@@ -270,33 +278,46 @@ func (i *collectionSimpleIndex) deleteDocIndex(
 	return i.deleteIndexKey(ctx, txn, key)
 }
 
+// hasIndexKeyNilField returns true if the index key has a field with nil value
+func hasIndexKeyNilField(key *core.IndexDataStoreKey) bool {
+	const cborNil = 0xf6
+	for i := range key.FieldValues {
+		if len(key.FieldValues[i]) == 1 && key.FieldValues[i][0] == cborNil {
+			return true
+		}
+	}
+	return false
+}
+
 type collectionUniqueIndex struct {
 	collectionBaseIndex
 }
 
 var _ CollectionIndex = (*collectionUniqueIndex)(nil)
 
-func (i *collectionUniqueIndex) Save(
+func (i *collectionUniqueIndex) saveRecord(
```

Comment:
nitpick: Why `saveRecord` and not `save`?

Reply:
no parcilar reason. Changed